services/datcore-adapter/src/simcore_service_datcore_adapter/modules/pennsieve.py

```diff
@@ -0,0 +1,378 @@
+        )
+
+        return (
+            [
+                DatasetMetaData(
+                    id=d["content"]["id"],
+                    display_name=d["content"]["name"],
+                )
+                for d in dataset_page["datasets"]
+            ],
+            dataset_page["totalCount"],
+        )
+
+    async def list_packages_in_dataset(
+        self,
+        api_key: str,
+        api_secret: str,
+        dataset_id: str,
+        limit: int,
+        offset: int,
+    ) -> Tuple[Sequence, Total]:
+        dataset_pck = await self._get_dataset(api_key, api_secret, dataset_id)
+        return (
+            [
+                FileMetaData.from_pennsieve_package(
+                    pck,
+                    await self._get_package_files(
+                        api_key, api_secret, pck["content"]["nodeId"]
+                    )
+                    if pck["content"]["packageType"] != "Collection"
+                    else [],
+                    base_path=Path(dataset_pck["content"]["name"]),
+                )
+                for pck in islice(dataset_pck["children"], offset, offset + limit)
+            ],
+            len(dataset_pck["children"]),
+        )
+
+    async def list_packages_in_collection(
+        self,
+        api_key: str,
+        api_secret: str,
+        dataset_id: str,
+        collection_id: str,
+        limit: int,
+        offset: int,
+    ) -> Tuple[Sequence, Total]:
+        dataset = await self._get_dataset(api_key, api_secret, dataset_id)
+        collection_pck = await self._get_package(api_key, api_secret, collection_id)
+        # compute base path ancestors are ordered
+        base_path = Path(dataset["content"]["name"]) / (
+            Path(
+                "/".join(
+                    [
+                        ancestor_pck["content"]["name"]
+                        for ancestor_pck in collection_pck["ancestors"]
+                    ]
+                )
+            )
+            / collection_pck["content"]["name"]
+        )
+
+        return (
+            [
+                FileMetaData.from_pennsieve_package(
+                    pck,
+                    await self._get_package_files(
+                        api_key, api_secret, pck["content"]["nodeId"]
+                    )
+                    if pck["content"]["packageType"] != "Collection"
+                    else [],
+                    base_path=base_path,
+                )
+                for pck in islice(collection_pck["children"], offset, offset + limit)
+            ],
+            len(collection_pck["children"]),
+        )
+
+    async def list_all_dataset_files(
+        self, api_key: str, api_secret: str, dataset_id: str
+    ) -> List:
+        """returns ALL the files belonging to the dataset, can be slow if there are a lot of files"""
+
+        async def _parse_dataset_items() -> List[FileMetaData]:
+            file_meta_data = []
+            cursor = ""
+            page_size = 1000
```

Comment:
is this to avoid pagination? why 1000?

Reply:
no this is the code as it was before in storage service while using the blackfynn API. I did not change it.

Since this is not scalable, to use it this way. this is also the route called ```list_dataset_files_legacy``` that brings us there. this should disappear if we want to handle the real datcore datasets.

Reply:
maybe just put a comment on it stating the above, thanks